### ISSUE TITLE
inconsistent arguments count is not detected

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#259]
Imagine this EO program:

```
[] > foo
  bar 42 > x
  bar 1 2 3 > y
```

It's illegal, because two objects with the same `@base` have different number of arguments.

We should check this on the level of `Programs`, not a single XMIR file.

This should lead to a warning, not error.

**Comment by maxonfjvipon** [comment]
@yegor256 what about partial application?

**Comment by yegor256** [comment]
@maxonfjvipon it's possible, but very rarely. That's why it's a warning. When someone will use partial application, they will add `+unlint` to the program.

**Comment by h1alexbel** [comment]
@yegor256 I'm in

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#365](https://github.com/objectionary/lints/issues/365) is still not solved.

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#502](https://github.com/objectionary/lints/issues/502) is still not solved; solved: [#365](https://github.com/objectionary/lints/issues/365).

**Comment by 0pdd** [comment]
@yegor256 all 2 puzzles are solved here: [#365](https://github.com/objectionary/lints/issues/365), [#502](https://github.com/objectionary/lints/issues/502).
